bitbots_vision/src/bitbots_vision/vision_modules/color.py

```diff
@@ -173,64 +180,48 @@ def mask_image(self, image):
+
+    def __init__(self, debug_printer, package_path, config):
         # type:(DebugPrinter, str, dict, bool) -> None
         """
-        PixelListColorDetector is a ColorDetector, that is based on a color space.
-        The color space is initially loaded from color-space-file at color_path (in config)
-        and optionally adjustable to changing color conditions (dynamic color space).
-        The color space is represented by boolean-values for RGB-color-values.
+        Initiating of PixelListColorDetector.
 
         :param DebugPrinter debug_printer: debug-printer
         :param str package_path: path of package
-        :param dict vision_config: vision config
-        :param bool primary_detector: true if is primary color detector
-            (only detector held by vision should be True) (Default: False)
+        :param dict config: vision config
         :return: None
         """
-        ColorDetector.__init__(self, debug_printer)
+        super(PixelListColorDetector, self).__init__(debug_printer)
         self.bridge = CvBridge()
 
-        self.vision_config = vision_config
-
-        self.primary_detector = primary_detector
+        self.config = config
 
         # concatenate color-path to file containing the accepted colors of base color space
-        if self.vision_config['vision_use_sim_color']:
-            self.color_path = package_path + self.vision_config['field_color_detector_path_sim']
+        if self.config['vision_use_sim_color']:
+            self.color_path = package_path + self.config['field_color_detector_path_sim']
         else:
-            self.color_path = package_path + self.vision_config['field_color_detector_path']
-        self.base_color_space = self.init_color_space(self.color_path)
-        self.color_space = np.copy(self.base_color_space)
+            self.color_path = package_path + self.config['field_color_detector_path']
 
-        # toggle publishing of mask_img msg
-        self.publish_mask_img_msg = self.vision_config['vision_mask_img_msg']
-        
-        # toggle publishing of mask_img_dyn msg with dynamic color space
-        self.publish_mask_img_dyn_msg = self.vision_config['dynamic_color_space_mask_img_dyn_msg']
-
-        # toggle use of dynamic color space
-        self.dynamic_color_space_turned_on = self.vision_config['dynamic_color_space']
-
-        # Subscribe to 'color_space'-messages from DynamicColorSpace
-        self.color_space_subscriber = rospy.Subscriber(
-            'color_space',
-            ColorSpace,
-            self.color_space_callback,
-            queue_size=1,
-            buff_size=2**20)
-    
         # Register publisher for 'mask_image'-messages
         self.imagepublisher = rospy.Publisher(
-            "/mask_image",
-            Image,
-            queue_size=1)
-
-        # Register publisher for 'mask_image_dyn'-messages
-        self.imagepublisher_dyn = rospy.Publisher(
-            "/mask_image_dyn",
+            "/field_mask",
```

Comment:
this topic could be configurable

Reply:
closed with 1d1bfc5babf81c18a5b762e4508d002de9d94d5c and c7c947b904fc75ac6834c9e8ae846a67bbeb3df5